miso-web/src/main/webapp/scripts/box_visualization.js

```diff
@@ -20,686 +20,696 @@
         }
-      }
-    });
-    jQuery('#dialogDialog').dialog('open');
-  };
+      });
 
-  return self;
-};
-
-Box.ScanDiff = function() {
-  var self = new BoxVisual();
-
-  self.show = function(results) {
-    self.results = results;
-
-    var diffs = results.diffs.map(function(d) {
-      switch (d.action) {
-      case 'added':
-        return '<li style="color:green;"><b>+</b> ' + d.modified.name + ' added to the box at position ' + d.coordinates + '</li>';
-      case 'removed':
-        return '<li style="color:red;"><b>-</b> ' + d.original.name + ' removed from ' + d.coordinates + '</li>';
-      case 'changed':
-        return '<li style="color:orange;"><b>!</b> ' + d.original.name + ' replaced by ' + d.modified.name + ' at ' + d.coordinates
-            + '</li>';
-      default:
-        return '<li><b>?</b> Unknown change at ' + (d.coordinates || 'unknown position') + '</li>';
-      }
-    });
-
-    var message = (results.errors.length == 0 ? "<h1>Scan Success! </h1>" : '<h1 class="warning">Scan Failed!</h1>')
-        + results.errors.map(function(err) {
-          return "<p>Position " + err.coordinates + ": " + err.message + "</p>";
-        }).join("") + (diffs.length == 0 ? "" : "<p>Box has changed. See below.</p>");
-
-    jQuery('#dialogInfoAbove').html(message);
-    jQuery('#dialogInfoBelow').html(
-        "<ul style=\"list-style-type: none;overflow:hidden; overflow-y:scroll;height:125px;\">" + diffs.join("") + '</ul>');
-
-    jQuery('#dialogDialog').dialog({
-      autoOpen: false,
-      title: 'Scan Results',
-      width: Box.dialogWidth,
-      modal: true,
-      resizable: false,
-      position: [jQuery(window).width() / 2 - Box.dialogWidth / 2, 50],
-      buttons: {
-        "Save": function() {
-          if (results.errors.length == 0 || confirm("Do you want to save changes even though there are scanning errors?")) {
-            Box.saveContents(self.results.items);
+      var message = (results.errors.length == 0 ? "<h1>Scan Success! </h1>" : '<h1 class="warning">Scan Failed!</h1>')
+          + results.errors.map(function(err) {
+            return "<p>Position " + err.coordinates + ": " + err.message + "</p>";
+          }).join("") + (diffs.length == 0 ? "" : "<p>Box has changed. See below.</p>");
+
+      jQuery('#dialogInfoAbove').html(message);
+      jQuery('#dialogInfoBelow').html(
+          "<ul style=\"list-style-type: none;overflow:hidden; overflow-y:scroll;height:125px;\">" + diffs.join("") + '</ul>');
+
+      jQuery('#dialogDialog').dialog({
+        autoOpen: false,
+        title: 'Scan Results',
+        width: Box.dialogWidth,
+        modal: true,
+        resizable: false,
+        position: [jQuery(window).width() / 2 - Box.dialogWidth / 2, 50],
+        buttons: {
+          "Save": function() {
+            if (results.errors.length == 0 || confirm("Do you want to save changes even though there are scanning errors?")) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this usage of confirm(...). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)